### ISSUE TITLE
Large discrepancy of OmniBench metrics in the paper

### ISSUE THREAD

**Issue by volcverse** [QwenLM/Qwen2.5-Omni#171]
Hi, thank you for releasing the code and models!

I tried reproducing the results reported in the paper on the OmniBench benchmark using the released model. However, I noticed that the performance across several tasks is significantly lower than what’s reported in the paper in Speech and Sound Event.

Could you kindly clarify that is the released model the exact one used for the paper results? Or was a stronger closed-source model used to produce the metrics in the paper? Additionally, would it be possible to release the evaluation code for OmniBench?

Here's my results:

**Speech**
My: 37.74%    Paper: 55.25%

**Music**
My: 53.77%    Paper: 52.83%

**Sound Event**
My: 47.92%    Paper: 60.00%

Any response will be greatly appreciated! :)

**Comment by zzchust** [comment]
M

**Comment by volcverse** [comment]
> Yes, based on my tests, I found that the metrics differ significantly from those in the paper. When both audio and images are used as input, the average accuracy is 0.46.

Yes. My avg accuracy is also around 0.47.

**Comment by wangxiongts** [comment]
Hi, we used thinker_do_sample=False and repetition_penalty=1.0 on all evaluation tasks except for the two collections of voicbench: alpaca and common eval, the default option in transformers code for thinker_do_sample is true.

**Comment by Alex-Songs** [comment]
@wangxiongts Is the system prompt using the default setting, or is it set to "You are a helpful assistant"?

**Comment by Alex-Songs** [comment]
@wangxiongts We set thinker_do_sample=False and repetition_penalty=1.0, but the final average accuracy is 0.4597(525/1142), which is significantly different from the results in the paper. Could you provide a reproducible evaluation script?

**Comment by Alex-Songs** [comment]
@wangxiongts  This is my codes:

`import soundfile as sf

import torch, os, json, re

from transformers import Qwen2_5OmniModel, Qwen2_5OmniProcessor
from qwen_omni_utils import process_mm_info
from tqdm import tqdm

from answer_parsing import parse_multi_choice_response


def split_string_by_options(input_string):
    pattern = r'(A\..*?)(B\..*?)(C\..*?)(D\..*)'
    matches = re.findall(pattern, input_string, re.DOTALL)
    return [match.strip() for match in matches[0]]


# default: Load the model on the available device(s)
model = Qwen2_5OmniModel.from_pretrained("Qwen2.5-Omni-7B", torch_dtype=torch.bfloat16,  device_map="auto",  attn_implementation="flash_attention_2")
processor = Qwen2_5OmniProcessor.from_pretrained("Qwen2.5-Omni-7B")


input_file="OmniBench.jsonl"
data_list=[]
with open(input_file, 'r', encoding='utf-8') as infile:
    for line in infile:
        data = json.loads(line)
        data_list.append(data)

all_choices = ['A', 'B', 'C', 'D']

results_dict_list=[]
for one_data in tqdm(data_list):

    question=one_data['question']
    options = one_data['opition']
    prompt = f"Please answer the following question based on the given image and audio:\n{question}.\nPlease choose only one answer from the following options:\n{options}."

    all_options =options
    index2ans={}
    for j in range(len(all_choices)):
        index2ans[all_choices[j]] = all_options[j].replace(f"{all_choices[j]}.", "").strip()


    conversation = [
        {
            "role": "system",
            "content": "You are a helpful assistant.",
        },
        {
            "role": "user",
            "content": [
                {"type": "text", "text": prompt},
                {"type": "image", "image": one_data['image_path']},
                {"type": "audio", "audio": one_data['audio_path']},
            ],
        },
    ]

    # set use audio in video
    USE_AUDIO_IN_VIDEO = True

    # Preparation for inference
    text = processor.apply_chat_template(conversation, add_generation_prompt=True, tokenize=False)
    audios, images, videos = process_mm_info(conversation, use_audio_in_video=USE_AUDIO_IN_VIDEO)
    inputs = processor(text=text, audios=audios, images=images, videos=videos, return_tensors="pt", padding=True, use_audio_in_video=USE_AUDIO_IN_VIDEO)
    inputs = inputs.to(model.device).to(model.dtype)

    text_ids = model.generate(**inputs, use_audio_in_video=USE_AUDIO_IN_VIDEO, return_audio=False, thinker_do_sample=False, repetition_penalty=1.0)
    text_ids = text_ids[:, inputs.input_ids.size(1) :]
    text = processor.batch_decode(text_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False)[0]


    parsed_response = parse_multi_choice_response(text, all_choices, index2ans)
    correct_answer = parse_multi_choice_response(one_data['answer'], all_choices, index2ans)


    results_dict = {
        "index": one_data['index'],
        "audio_type": one_data['audio_type'], 
        "question": question,
        "options": options,
        "response": text,
        "answer":one_data['answer'],
        "parsed_response": parsed_response,
        "correct_answer": correct_answer,
        "is_correct": parsed_response == correct_answer
    }

    results_dict_list.append(results_dict)

    # Calculate accuracy
    correct_count = sum(result['is_correct'] for result in results_dict_list if result)
    accuracy = correct_count / len(results_dict_list)
    print(f"Accuracy: {accuracy:.4f}, {correct_count}/{len(results_dict_list)}")
    print("\n", flush=True)`

**Comment by wangxiongts** [comment]
During our evaluation in OmniBench, we find in https://github.com/multimodal-art-projection/OmniBench/blob/8979cdd1199a95b9217afacdac8b2d2fd7c20d53/inference/calculate_metrics.py#L75, the author use `parse_multi_choice_response` to compute correct_answer, but some answers like "C. A man is ...." will be parsing as choice "A", so we still used this script to ensure that the evaluation result is on fair with other open source results. But in fact, our real score after fixing this problem should be around 0.49 in avg.

**Comment by Alex-Songs** [comment]
@wangxiongts  I haven't modified the scoring script for OmniBench, and indeed many options were parsed as A. In this situation, the average accuracy we measured was 0.4597.

**Comment by Alex-Songs** [comment]
@volcverse Hi, have you reproduced the results?

**Comment by wangxiongts** [comment]
> [@wangxiongts](https://github.com/wangxiongts) I haven't modified the scoring script for OmniBench, and indeed many options were parsed as A. In this situation, the average accuracy we measured was 0.4597.

Have you ensured the choice `A....B....C...D` in `options = one_data['opition']`, due to the non-standard format of some data, it does not include the option ABCD.  And conversation should be
```python
            {"type": "image", "image": one_data['image_path']},
            {"type": "audio", "audio": one_data['audio_path']},
            {"type": "text", "text": prompt},
```

**Comment by Alex-Songs** [comment]
@wangxiongts Yes, I manually added A, B, C, D to the options and answers.

**Comment by Alex-Songs** [comment]
@wangxiongts I replace the conversation, and the test results is 0.4869, which still has a significant gap compared to the paper.

**Comment by wangxiongts** [comment]
I think this is due to error fluctuations, which may cause accuracy errors different from our test GPU and system environment.

**Comment by wangxiongts** [comment]
I think we only need to ensure that the option resolution is correct and that it will not be affected by cases like "C. A man is .... ". As I replied before, the expected score should be 0.49.

**Comment by HarryHsing** [comment]
> [@wangxiongts](https://github.com/wangxiongts) I replaced the conversation, and the test results show an accuracy of 0.5595, which still has some gap compared to the paper.
> 
> Accuracy, Invalid Rate 0.5595, 0.0000 Stats by audio type: music total: 106 Correct, Invalid, Wrong Response Rate: music: 0.5755, 0.0000, 0.4245 speech total: 771 Correct, Invalid, Wrong Response Rate: speech: 0.5525, 0.0000, 0.4475 sound event total: 265 Correct, Invalid, Wrong Response Rate: sound event: 0.5736, 0.0000, 0.4264

Hi Alex, may i know how you reproduce this result in the end? thansk a lot for your help!

**Comment by gengenkai** [comment]
> [@wangxiongts](https://github.com/wangxiongts) I replaced the conversation, and the test results show an accuracy of 0.5595, which still has some gap compared to the paper.
> 
> Accuracy, Invalid Rate 0.5595, 0.0000 Stats by audio type: music total: 106 Correct, Invalid, Wrong Response Rate: music: 0.5755, 0.0000, 0.4245 speech total: 771 Correct, Invalid, Wrong Response Rate: speech: 0.5525, 0.0000, 0.4475 sound event total: 265 Correct, Invalid, Wrong Response Rate: sound event: 0.5736, 0.0000, 0.4264

Could you share you eval script？ Maybe different from the github official repo? Thanks.

**Comment by gengenkai** [comment]
> I think we only need to ensure that the option resolution is correct and that it will not be affected by cases like "C. A man is .... ". As I replied before, the expected score should be about 0.49.

Could you share your eval code  and metric_compute script? Most of the people who use your released ckpt and official eval script can not reproduce 0.55 or 0.49 . Is it any trick on computing the evaluation metric ?

**Comment by wudongming97** [comment]
> OmniBench.jsonl

Hi, how can I obtain the OmniBench.jsonl file for evaluation? Thanks very much!
